api/uaa/refresh_token.go

```diff
@@ -36,6 +51,7 @@ func (client *Client) RefreshAccessToken(refreshToken string) (RefreshToken, err
 			"Content-Type": {"application/x-www-form-urlencoded"},
 		},
 		Body: body,
+		AddClientAuthorization: true,
 	})
```

Comment:
`request.SetBasicAuth()` does not actually do what [section 2.3.1 of the OAuth 2.0 spec](https://tools.ietf.org/html/rfc6749#section-2.3.1) requires. ie. the client ID and secret must be URL encoded, before being constructed into a basic auth header, which is why I added the `oauthClientAuthorizationHeader()` method.